lib/theme_check/shopify_liquid/source_index.rb

```diff
@@ -3,50 +3,52 @@
 require 'json'
 require 'pathname'
 
-require_relative 'source_index/base_entry'
-require_relative 'source_index/filter_entry'
-require_relative 'source_index/object_entry'
-require_relative 'source_index/parameter_entry'
-require_relative 'source_index/property_entry'
-require_relative 'source_index/return_type_entry'
-require_relative 'source_index/tag_entry'
-
-require_relative './source_manager'
-
 module ThemeCheck
   module ShopifyLiquid
     class SourceIndex
       class << self
         def filters
-          @filters ||= load_file(:filters)
-            .map { |hash| SourceIndex::FilterEntry.new(hash) }
+          @filters = nil if FilterState.outdated?
+
+          @filters ||= FilterState.mark_up_to_date &&
+            load_file(:filters)
+              .map { |hash| FilterEntry.new(hash) }
         end
 
         def objects
-          @objects ||= load_file(:objects)
-            .concat(built_in_objects)
-            .map { |hash| SourceIndex::ObjectEntry.new(hash) }
+          @objects = nil if ObjectState.outdated?
+
+          @objects ||= ObjectState.mark_up_to_date &&
+            load_file(:objects)
+              .concat(built_in_objects)
+              .map { |hash| ObjectEntry.new(hash) }
         end
 
         def tags
-          @tags ||= load_file(:tags)
-            .map { |hash| SourceIndex::TagEntry.new(hash) }
+          @tags = nil if TagState.outdated?
+
+          @tags ||= TagState.mark_up_to_date &&
+            load_file(:tags)
+              .map { |hash| TagEntry.new(hash) }
         end
 
         private
 
         def load_file(file_name)
-          read_json(SourceManager.local_path(file_name))
+          read_json(local_path(file_name))
         end
 
-        def read_json(path)
-          SourceManager.download_or_refresh_files
+        def local_path(file_name)
```

Comment:
`SourceManager.local_path` only builds the path to `file_name`.
`local_path` in _this_ class however also might download some files.
Therefore, I would rename this method.

Reply:
I think that `local_path!` is a better name :+1:

I was just wondering if we could write
```
def load_file(file_name)
  SourceManager.download unless SourceManager.has_required_files?
  read_json(SourceManager.local_path(file_name))
end
```
instead and remove the `local_path!` method?

Or do we need it for testing purposes?